include/aspect/utilities.h

```diff
@@ -130,6 +130,8 @@ namespace aspect
      *   previous call to this function) then this vector is used as
      *   expected structure of the input parameter, and it is checked that
      *   @p key_value_map fulfills this structure.
+     * @param [in] allow_empty_keys Whether to allow empty keys that have
+     *   no associated entries. This also allows a completely empty map.
```

Comment:
I find this name and description not quite clear, and related to that, it's not quite clear to me what we actually want to allow here. Allow empty key sounds like 'allow no values per key'. But that's not quite what this is, correct? Here, someone can just leave out one of the compositional fields from their list in the parameter file? Something like `background:1, field1:1, field3:1` (leaving out field2). Then our list of keys does have a different number of entries than `nfields`. But we still do all of the checks, like, if you have just one entry in the list and more than one field, the keyword has to be `all`. Or do you just want to allow something like `field:` (specifying the key but not the value)?

It works for a completely empty map, but just having some of the keys specified makes this logic even more complicated. (I am not saying you should take this out, I get that we want to allow this for users, I just find it very hard to make sure all the logic is correct)